docs/reference/api-conventions.asciidoc

```diff
@@ -382,10 +382,14 @@ supported units are:
 
 [[size-units]]
 [float]
-=== Data size units
+=== Unit-less quantities
 
-Whenever the size of data needs to be specified, the value must specify the unit,
-like `10m` for 1 000 000 elements. The supported units are:
+Unit-less quantities means that they don't have a "unit" like "bytes" or "Hertz" or "meter" or "long tonne".
+
+Whenever the size of data needs to be specified or printed, the value may specify the multiplier like `10m` for
+1 000 000 elements. For example a "document count" or "queue depth" could be expressed using this notation.
```

Comment:
s/1 000 000/10 000 000/ ?

Also, I don't know if we are consistent, but I think we'd use "10,000,000". That is the normal "American" formatting but I think I see it all over our docs.

I might rewrite the second paragraph like
If one of these quantities is large we'll print it out like `10m` for `10,000,000` or `7k` for `7,000`. We'll still print `87` when we mean `87` though. These are the supported multipliers:

Or something like that. I dunno.